database.cc

```diff
@@ -876,13 +876,25 @@ void database::add_column_family(keyspace& ks, schema_ptr schema, column_family:
     schema = local_schema_registry().learn(schema);
     schema->registry_entry()->mark_synced();
 
+    auto repair_based_tombstone_gc = [this, ks_name = schema->ks_name()] {
+        // We enable repair based tombstones gc for all tables except tables
+        // with local replication strategy and tables with RF one.
+        auto& ks = find_keyspace(ks_name);
+        auto& rs = ks.get_replication_strategy();
+        auto erm = ks.get_effective_replication_map();
+        bool use_repair_based_tombstone_gc = rs.get_type() != locator::replication_strategy_type::local
+                && erm->get_replication_factor() != 1;
+        return use_repair_based_tombstone_gc;
+    };
+
     lw_shared_ptr<column_family> cf;
     if (cfg.enable_commitlog && _commitlog) {
        cf = make_lw_shared<column_family>(schema, std::move(cfg), *_commitlog, *_compaction_manager, *_cl_stats, _row_cache_tracker);
     } else {
        cf = make_lw_shared<column_family>(schema, std::move(cfg), column_family::no_commitlog(), *_compaction_manager, *_cl_stats, _row_cache_tracker);
     }
     cf->set_durable_writes(ks.metadata()->durable_writes());
+    cf->set_repair_based_tombstone_gc_func(std::move(repair_based_tombstone_gc));
```

Comment:
feels more natural to move this into table ctor.

Reply:
I wanted to avoid modifying 1000 places where we create the table object.  Also, I am thinking about adding a db reference in the table object so that we can avoid set the function at all. It's better to have a followup to reduce the noise.

Reply:
I feel like it's better to avoid db reference in table object, as db is one layer above table. A db ref in table means layer violation. I think it's better to either go with this callback set function or a callback passed to the table ctor.